src/status_im2/subs/communities.cljs

```diff
@@ -209,14 +209,36 @@
         (sort-by :position)
         (into []))))
 
-(defn reduce-over-categories
-  [community-id
-   joined
+(defn- get-chat-token-permissions
+  [{community-id :id permissions :token-permissions} chat-id]
+  (let [chat-composite-id (str community-id chat-id)]
+    (->> permissions
+         (map second)
+         (filter (fn [{composite-ids :chat_ids}]
+                   (some (fn [composite-id]
+                           (= composite-id chat-composite-id))
+                         composite-ids))))))
```

Comment:
if `permissions` is a map, we could use `vals` instead of `(map second)` to get the list of values.

Reply:
Unfortunately it's not a map @ulisesmac, it's a tuple permission id and the permission.